presto-main-base/src/main/java/com/facebook/presto/sql/planner/GroupedExecutionTagger.java

```diff
@@ -166,6 +167,15 @@ public GroupedExecutionTagger.GroupedExecutionProperties visitMergeJoin(MergeJoi
             // TODO: This will break the other use case for merge join operating on sorted tables, which requires grouped execution for correctness.
             return GroupedExecutionTagger.GroupedExecutionProperties.notCapable();
         }
+
+        if (isPrestoOnSpark) {
```

Comment:
When presto on spark is the environment, if one side is grouped execution capable, even if the other side is not, the data will still be partitioned by buckets, and have a bucket by bucket execution in presto on spark